packages/office-ui-fabric-react/src/components/Nav/Nav.tsx

```diff
@@ -209,44 +212,47 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
 
 // A tag used for resolving links.
 let _urlResolver;
-function _isLinkSelected(link: INavLink, selectedKey: string): boolean {
-    if (selectedKey && link.key === selectedKey) {
-      return true;
-    }
+function _isLinkSelected(link: INavLink, clickSelectedKey: string, propsSelectedKey: string): boolean {
+  // if caller passes in selectedKey, use it as first choice or
+  // if current state.selectedKey (from addressbar) is match to the link
+  if (propsSelectedKey !== undefined && link.key === propsSelectedKey ||
```

Comment:
This seems like it could be significantly better designed. 

1. `_isLinkSelected` could be a private. Then you can access props and state.
2. You can immediately use this.props.selectedKey if it is not undefined.
3. You can fall back to internal state.